src/helpers/contexts/GlobalNodeState.tsx

```diff
@@ -178,28 +173,23 @@ export const GlobalProvider = ({
     schemata,
     reactFlowWrapper,
 }: React.PropsWithChildren<GlobalProviderProps>) => {
-    const { useSnapToGrid } = useContext(SettingsContext);
+    const useSnapToGrid = useContextSelector(SettingsContext, (c) => c.useSnapToGrid);
 
-    const [nodes, setNodes, onNodesChange] = useNodesState<NodeData>([]);
-    const [edges, setEdges, onEdgesChange] = useEdgesState<EdgeData>([]);
```

Comment:
Does everything still work without the onNodes/EdgesChange functions?

Reply:
They weren't used at all. That's why I removed them.

Reply:
Oh, I guess the ones in ReactFlowBox do all the work. Makes sense.